src/daemon/Display.cpp

```diff
@@ -84,40 +84,45 @@ namespace SDDM {
         return VirtualTerminal::setUpNewVt();
     }
 
-    Display::Display(Seat *parent) : QObject(parent),
+    Display::Display(Seat *parent, std::optional<DisplayServerType> forceServerType)
+        : QObject(parent),
         m_auth(new Auth(this)),
         m_seat(parent),
         m_socketServer(new SocketServer(this)),
-        m_greeter(new Greeter(this)) {
-
-
+        m_greeter(new Greeter(this))
+    {
         // Save display server type
-        const QString &displayServerType = mainConfig.DisplayServer.get().toLower();
-        if (displayServerType == QStringLiteral("x11-user")) {
-            m_displayServerType = X11UserDisplayServerType;
-            m_terminalId = fetchAvailableVt();
-        } else if (displayServerType == QStringLiteral("wayland")) {
-            m_displayServerType = WaylandDisplayServerType;
-            m_terminalId = fetchAvailableVt();
+        if (forceServerType.has_value()) {
+            qDebug() << "Forcing display server type" << *forceServerType;
+            m_displayServerType = *forceServerType;
         } else {
-            if (displayServerType != QLatin1String("x11")) {
-                qWarning("\"%s\" is an invalid value for General.DisplayServer: fall back to \"x11\"",
-                     qPrintable(displayServerType));
+            const QString &displayServerType = mainConfig.DisplayServer.get().toLower();
```

Comment:
If we move this code to seat, we can simplify it and get rid of optionals

i.e
```
Display::Display(Seat *parent, DisplayServerType serverType) {
  m_displayServerType = serverType
  .. rest of the code as is...

Seat::createDisplay() {
    const QString &displayServerType = mainConfig.DisplayServer.get().toLower();
    if (displayServerType == QStringLiteral("x11-user"))
       return createDisplay(X11UserDisplayServerType);
       
}

Seat::createDisplay(DisplayServerType serverType)
as-is

```

Reply:
Not sure it's simpler, but there you have it.